storage/replica_command.go

```diff
@@ -1648,7 +1648,7 @@ func (r *Replica) applyNewLeaseLocked(
 			// in on the Lease struct.
```

Comment:
Add some commentary here. Something like:

>
The introduction of lease transfers implies that the previous
lease may have been shortened and we are now applying a
formally overlapping lease (since the old lease holder has
promised not to serve any more requests, this is kosher).
This means that we can't use the old lease's expiration but
instead use the new lease's Start to initiate the timestamp
cache.